model_analyzer/config/config_object.py

```diff
@@ -84,29 +85,44 @@ def set_value(self, value):
         schema = self._schema
 
         if type(value) is dict:
-            for key, value_ in value.items():
-                if key in schema:
-                    new_item = deepcopy(schema[key])
-
-                # If the key is not available in the schema, but wildcard is
-                # present, we use the schema for the wildcard.
-                elif '*' in schema:
-                    new_item = deepcopy(schema['*'])
-                else:
-                    return ConfigStatus(
-                        CONFIG_PARSER_FAILURE, f'Key "{key}" should not be '
-                        f'specified in field "{self.name()}".', self)
-
-                new_item.set_name(f'{self.name()}.{key}')
-                new_value[key] = new_item
-
-                # Set the value of the new field
-                config_status = new_item.set_value(value_)
-
-                # If it was not able to set the value, for this
-                # field, we fail.
-                if config_status.status() == CONFIG_PARSER_FAILURE:
-                    return config_status
+            if '**' in schema:
```

Comment:
What are you actually doing here? I think you need a higher level comment. The description does not seem to match needing to "deep copy schemas"

Reply:
Why did you need to move `*` to the top of the for loop?